### PR TITLE
Take while and drop while debug

### PR DESCRIPTION
**Pull request by ahronl**

### CODE REVIEW

File: src/mate_clj/core.clj [hunk 1]
```diff
@@ -156,3 +156,50 @@
...
    (dfilter (complement pred) coll)))
+
+(defn dtake-while
+  ([pred]
+     (fn [rf]
+       (fn
+         ([] (rf))
+         ([result] (rf result))
+         ([result input]
+            (println pred input " => " (pred input))
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: remove spaces from `" => "` to be `"=>"`

File: src/mate_clj/core.clj [hunk 2]
```diff
@@ -156,3 +156,50 @@
...
+              (rf result input)
+              (reduced result))))))
+  ([pred coll]
+    (lazy-seq
+      (when-let [s (seq coll)]
+        (println pred (first s) " => " (pred (first s)))
+        (when (pred (first s))
+          (cons (first s) (dtake-while pred (rest s))))))))
+
+
```
Comment (dborinsk) [inline src/mate_clj/core.clj:177]: remove extra line space

File: src/mate_clj/core.clj [hunk 3]
```diff
@@ -156,3 +156,50 @@
...
+         ([result] (rf result))
+         ([result input]
+            (println pred input " => " (pred input))
+            (if (pred input)
+              (rf result input)
+              (reduced result))))))
+  ([pred coll]
+    (lazy-seq
+      (when-let [s (seq coll)]
+        (println pred (first s) " => " (pred (first s)))
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: remove spaces from `" => "` to be `"=>"`

File: src/mate_clj/core.clj [hunk 4]
```diff
@@ -156,3 +156,50 @@
...
+(defn ddrop-while
+  ([pred]
+     (fn [rf]
+       (let [dv (volatile! true)]
+         (fn
+           ([] (rf))
+           ([result] (rf result))
+           ([result input]
+              (let [drop? @dv]
+	 	(println pred input " => " (pred input))
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: remove spaces from `" => "` to be `"=>"`

File: src/mate_clj/core.clj [hunk 5]
```diff
@@ -156,3 +156,50 @@
...
+                (if (and drop? (pred input))
+                  result
+                  (do
+                    (vreset! dv nil)
+                    (rf result input)))))))))
+  ([pred coll]
+     (let [step (fn [pred coll]
+                  (let [s (seq coll)]
+		    (println pred (first s) " => " (pred (first s)))
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: remove spaces from `" => "` to be `"=>"`

File: src/mate_clj/core.clj [hunk 6]
```diff
@@ -156,3 +156,50 @@
   ([pred] (dfilter (complement pred)))
   ([pred coll]
    (dfilter (complement pred) coll)))
+
+(defn dtake-while
+  ([pred]
+     (fn [rf]
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: i think all this `fn` block has two extra spaces, it should be just under the `[`
and its relevant to all the functions below

**Review by dborinsk — CHANGES_REQUESTED** [review]
Great addition to the library! Thanks for your contribution!

File: src/mate_clj/core.clj (hunk 1 above)
Comment (ahronl) [reply src/mate_clj/core.clj]: done

File: src/mate_clj/core.clj (hunk 3 above)
Comment (ahronl) [reply src/mate_clj/core.clj]: done

File: src/mate_clj/core.clj (hunk 5 above)
Comment (ahronl) [reply src/mate_clj/core.clj]: done

File: src/mate_clj/core.clj (hunk 6 above)
Comment (ahronl) [reply src/mate_clj/core.clj]: done

Comment (dborinsk) [reply src/mate_clj/core.clj]: It is still not in the correct indentation check out the other functions in this file.
The block after the arities should be in the same line.

```clojure
(defn dtake-while
  ([pred]
   (fn [rf]
     (fn
       ([] (rf))
       ([result] (rf result))
       ([result input]
        (println pred input "=>" (pred input))
        (if (pred input)
          (rf result input)
          (reduced result))))))
  ([pred coll]
   (lazy-seq
     (when-let [s (seq coll)]
       (println pred (first s) "=>" (pred (first s)))
       (when (pred (first s))
         (cons (first s) (dtake-while pred (rest s))))))))
```

File: src/mate_clj/core.clj [hunk 7]
```diff
@@ -156,3 +156,49 @@
...
+              (rf result input)
+              (reduced result))))))
+  ([pred coll]
+    (lazy-seq
+      (when-let [s (seq coll)]
+        (println pred (first s) "=>" (pred (first s)))
+        (when (pred (first s))
+          (cons (first s) (dtake-while pred (rest s))))))))
+
+(defn ddrop-while
```
Comment (dborinsk) [inline src/mate_clj/core.clj]: Also here the indentation is not correct, it should be:
```clojure
(defn ddrop-while
  ([pred]
   (fn [rf]
     (let [dv (volatile! true)]
       (fn
         ([] (rf))
         ([result] (rf result))
         ([result input]
          (let [drop? @dv]
            (println pred input "=>" (pred input))
            (if (and drop? (pred input))
              result
              (do
                (vreset! dv nil)
                (rf result input)))))))))
  ([pred coll]
   (let [step (fn [pred coll]
                (let [s (seq coll)]
                  (println pred (first s) "=>" (pred (first s)))
                  (if (and s (pred (first s)))
                    (recur pred (rest s))
                    s)))]
     (lazy-seq (step pred coll)))))
```

and also in `dsplit-with`

**Review by dborinsk — APPROVED** [review]

**Pull request merged**